### PR TITLE
Put stylo's pseudo-class into a list file

### PR DESCRIPTION
**Pull request by upsuper**
r? @emilio

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/15153)

### CODE REVIEW

File: components/style/gecko/non_ts_pseudo_class_list.rs [hunk 1]
```diff
@@ -0,0 +1,47 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/*
+ * This file contains a helper macro includes all supported non-tree-structural
+ * pseudo-classes.
+ *
+ * This file is NOT INTENDED to be compiled as a standalone module.
```
Comment (emilio) [inline components/style/gecko/non_ts_pseudo_class_list.rs]: This is not autogenerated right? Please note it in this comment (potentially keeping the `FIXME` that you removed from the other file).

File: components/style/gecko/non_ts_pseudo_class_list.rs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ */
+
+pseudo_class_list! {
+    ("any-link", AnyLink, anyLink, _, _),
+    ("link", Link, link, _, _),
+    ("visited", Visited, visited, _, _),
+    ("active", Active, active, IN_ACTIVE_STATE, _),
+    ("focus", Focus, focus, IN_FOCUS_STATE, _),
+    ("fullscreen", Fullscreen, fullscreen, IN_FULLSCREEN_STATE, _),
+    ("hover", Hover, hover, IN_HOVER_STATE, _),
```
Comment (emilio) [inline components/style/gecko/non_ts_pseudo_class_list.rs]: I don't love the `gecko_type` name, but I don't have a better suggestion :)

**Review by emilio — APPROVED** [review]
Looks fine, r=me % comments.

Just one question, this is just in order to simplify stuff? or you plan to autogenerate/add more pseudo classes in the near future?

**Comment by upsuper** [comment]
I don't have plan to auto-generate it at this moment, since we have far fewer pseudo-classes than Gecko currently. But I guess this kind of list would be easier to auto-generate once we want to do so. Before that, I think it is still useful to have this list so that it's trivial to add support for many of the pseudo-classes.

**Comment by upsuper** [comment]
> LLVM ERROR: IO failure on output stream.

:/

**Pull request merged**